From f4e754e66e3661274df624bc328991cd88dd03d6 Mon Sep 17 00:00:00 2001
From: Thomas Friedrich <tfriedr@us.ibm.com>
Date: Fri, 3 Jul 2015 10:53:22 -0700
Subject: [PATCH] PARQUET-324: row count incorrect if data file has more than
 2^31 rows

Need to change numRows counter from int to long to account for input files with more than 2^31 rows.

Author: Thomas Friedrich <tfriedr@us.ibm.com>

Closes #233 from tfriedr/parquet-324 and squashes the following commits:

0120205 [Thomas Friedrich] change numRows from int to long
---
 .../parquet/format/converter/ParquetMetadataConverter.java      | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java b/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java
index e8b6d7384e..fdeb2ba246 100644
--- a/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java
+++ b/parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java
@@ -89,7 +89,7 @@ public class ParquetMetadataConverter {
   public FileMetaData toParquetMetadata(int currentVersion, ParquetMetadata parquetMetadata) {
     List<BlockMetaData> blocks = parquetMetadata.getBlocks();
     List<RowGroup> rowGroups = new ArrayList<RowGroup>();
-    int numRows = 0;
+    long numRows = 0;
     for (BlockMetaData block : blocks) {
       numRows += block.getRowCount();
       addRowGroup(parquetMetadata, rowGroups, block);
